### ISSUE TITLE
Hangs when compressing many files...

### ISSUE THREAD

**Issue by RDKRACZ** [Huluti/Curtail#94]
It stopped responding when it started working with ~150 files on my Intel i7-6700K PC.
Also, happens every time I try.

**Comment by Jacobthegr8** [comment]
Same here, I have given the duty of Curtail to losslessly compress my entire 200GB Nextcloud storage backup. It goes great for 10 seconds then the interface jams. It goes form using 16 threads to just one at %100. Props to your program for not crashing and closing outright. I have no clue how these programs are developed but it happens when I drag and drop a folder in Zoron OS. I guess it has trouble navigating all the directories and finding compressible items WHILE it is compressing. Maybe split the two processes and start compressing when there is no more directory finding to do? Hopefully there can be a Timeout of 30 seconds or something if It cant compress the image. Anyway, As it stands, I can still use your program doing batches at a time so thank you for your contribution to FOSS software. It really does help.

**Comment by Jacobthegr8** [comment]
On further inspection, When a folder is handed to the program by drag and drop, It can add all entities but the interface slows down when I scroll in the main view. The lag will still happen if there is no compression currently happening.

**Comment by Huluti** [comment]
Hello! Could you please retry with the lastest version (1.5.0)? I know it's been a long time but maybe you're still using Curtail? :)

**Comment by Jacobthegr8** [comment]
Still am! It never left my desktop. The lossless compression process feels and looks seamless! Looks like compressing more than one image at a time really choked the computer. I am on 1.5.0 from flathub. But a new problem happens, I am on Zorin OS (ubuntu) and now, dragging and dropping a folder does nothing. I can individually drag and drop photos from that folder and even select multiple pictures for compression. Keep up the good work though! The broken feature did exist in previous versions so you at least have something to work with. Keep up the good work! Cheers.

**Comment by Huluti** [comment]
Yes for the folder bug, caused by the Gtk4 migration. I will see how to resolve that soon :)
So no more hangs that's it?
Thank you for your support!

**Comment by Jacobthegr8** [comment]
I have the png lossless compression on its highest level. For my desktop screenshots, the 20 second timeout is reached. The error appears as such, Command 'optipng - clobber -o7 "DIRECTORY" -out "DIRECTORY" -preserve' timed out after 20 seconds. While there are no "hangs" the "timeout" warning stays frozen and cannot be dismissed until the compression que is finished. The compression does finish though. During compression, the system "Program not responding" prompt appears but auto closes when the next entity on the list gets displayed. TLDR, My laptop processor is not powerful enough to compress the images under 20 seconds. Some images do not receive a "file size shrink percent" if they time out. During the 500 lossless picture compression test including both PNG, JPG files and some other (non image) files. Many PNG files where skipped (timed out) and all JPG files where compressed with around %2 savings. I have played with the advanced tab so, after changing the PNG compression level to 4, around %20 savings where seen with the PNG's tested IF they did not time out (Half of the images compressed timed out).

TLDR, Some PNG files can compress before the 20 second timeout. A 500 image test was conducted, JPG compression was fine but PNG files where troublesome both at lossless compression levels 4 and 6.

**Comment by Jacobthegr8** [comment]
Thanks for making a good program great! When Curtail is working to its maximum potential ill hit it with 300GB of photos I've taken/downloaded through the years. It is a huge folder with many misc file extensions and many directories. I already have one friend excited to run it on one of his Syncthing folders. Like me and him, we have hundreds of Solidworks renders and scanned notes that REALLY cut down in size with lossless compression. Again, keep up the good work!

**Comment by Jacobthegr8** [comment]
Good news, after some more testing. Setting the PNG losless compression level 1 actually compresses PNG without timeouts. Now I get %20 savings on my images.

**Comment by Huluti** [comment]
@Jacobthegr8 should be really better with the latest version (1.6.0)! No more UI freeze, configurable timeout, compression in an other thread...

**Comment by Jacobthegr8** [comment]
WOW! Folders work, more advanced settings, configurable timeout and fine tuned lossless compression settings! Time to boot up my photo server! I cannot thank you enough! I have uninstalled the 2 other older lossless compression programs on my computer because THIS ONE IS IT! I cannot wait to feed curtail my hundreds of Solidworks renders! Solidworks just LOVES to make renders that can be compressed up to %50 lossless. Thank you so much for your efforts.

**Comment by Huluti** [comment]
Happy to hear that!! Finally ;) 
I close this issue so. Don't hesitate to reopen if needed!
